manifests/plugin/python.pp

```diff
@@ -1,19 +1,22 @@
 # See http://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_python
 class collectd::plugin::python (
   # Python 2 defaults to 'ascii' and Python 3 to 'utf-8'
-  $encoding            = undef,
-  $ensure              = 'present',
+  $encoding              = undef,
+  $ensure                = 'present',
   # Unlike most other plugins, this one should set "Globals true". This will cause collectd
   # to export the name of all objects in the Python interpreter for all plugins to see.
-  Boolean $globals     = true,
-  Boolean $interactive = false,
-  $interval            = undef,
-  Boolean $logtraces   = false,
-  $manage_package      = undef,
-  Array $modulepaths   = [],
-  Hash $modules        = {},
-  $order               = '10',
-  $conf_name           = 'python-config.conf',
+  Boolean $globals              = true,
+  Boolean $interactive          = false,
+  $interval                     = undef,
+  Boolean $logtraces            = false,
+  $manage_package               = undef,
+  Array $modulepaths            = [],
+  Hash $modules                 = {},
+  $order                        = '10',
+  $conf_name                    = 'python-config.conf',
+  Boolean $manage_python        = true,
```

Comment:
I'm currently unsure if this should default to true. This probably breaks the catalog for many people. Should it default to false and be documented in the README.md?

Reply:
+1 to not install/manage prerequisites by default.

Reply:
```
collectd::plugin::python
```

cannot work if python is not installed. What's the difference between a pre-requisite and a requirement?
Package is added with 'ensure_packages' to reduce impact to people catalogs.

Happy to change of course.

Reply:
I'll change because it's easier to do and get through. It's only ubunutu and I never use that so...

Think I already answered the python version comment above.

> ensure_packages works only if this class gets evaluated at the end.

The other class probably needs fixing if it does not ensure_resource on such a fundamental package.
Essentially this is saying it's never okay to install python but given ubunut 16 docker images don't have python this is a `new` necessity.

Of course there's a total alternate method to this. Remove the facter fact that depends on a python interpreter to work and just set sensible values for *python_sitelib* on different OSes.